buildserver-monitor/httpClient.cpp

```diff
@@ -0,0 +1,146 @@
+/**
+   \file httpClient.cpp
+
+   \licence "THE BEER-WARE LICENSE" (Revision 42):
+            <jilvin.wiegmus@fourtress.nl> wrote this file. As long as you retain
+            this notice you can do whatever you want with this stuff. If we
+            meet some day, and you think this stuff is worth it, you can buy me
+            a beer in return.
+                                                                  melip muskman
+
+   \brief   Http configuration file for the Buildserver Monitor.
+
+   \details Intended use is to provide an easier means to handle a http requests
+            and retrieve data for a specified buildserver URL.
+
+   \date    01-2020
+*/
+
+/************************************************************************/
+/* Includes                                                             */
+/************************************************************************/
+#include <string>
+#include "config.h"
+#include "http_config.h"
+#include "httpClient.hpp"
+#include <ArduinoJson.h>
+
+
+/************************************************************************/
+/* Constants                                                            */
+/************************************************************************/
+
+
+/************************************************************************/
+/* Public Methods                                                       */
+/************************************************************************/
+/**
+   \brief   Constructor.
+   \param   logger      Logging class.
+*/
+httpClient::httpClient(ILogging& logger) :
+  mLogger(logger)
+{
+  mInit = false;
+  mJsonString = "";
+}
+
+/**
+   \brief   Destructor, disconnects (if connected).
+*/
+httpClient::~httpClient()
+{
+
+}
+
+bool httpClient::Init()
+{
+  if (!mInit) {
+    mLogger.Log(LogLevel::INFO, "Initializing http client");
+    client = new BearSSL::WiFiClientSecure();
+    client->setFingerprint(SHA1_FINGERPRINT);
```

Comment:
Add check for nullptr, maybe device is out of resources. A log entry when that happens and failure result to enter the error state would be nice as well.